eio/client/dns_client_eio.ml

```diff
@@ -0,0 +1,204 @@
+type 'a env = <
+  clock         : Eio.Time.clock ;
+  net           : Eio.Net.t ;
+  fs            : Eio.Fs.dir Eio.Path.t ;
+  secure_random : Eio.Flow.source;
+  ..
+> as 'a
+
+type io_addr = Ipaddr.t * int
+type stack = {
+  sw            : Eio.Switch.t;
+  clock         : Eio.Time.clock;
+  net           : Eio.Net.t;
+  resolv_conf   : Eio.Fs.dir Eio.Path.t
+}
+
+module Transport : Dns_client.S 
+  with type io_addr = io_addr
+   and type stack   = stack
+   and type +'a io  = 'a  
+= struct 
+  type nonrec io_addr = io_addr
+  type nonrec stack = stack
+  type +'a io = 'a
+  type context = <Eio.Net.stream_socket; Eio.Flow.close>
+
+  type nameservers = 
+    | Static of io_addr Queue.t
+    | Resolv_conf of {
+        mutable nameservers : io_addr Queue.t ;
+        mutable digest      : Digest.t option ;
+      }
```

Comment:
what is the purpose of the Queue.t sprinkled here?

Reply:
`Queue.t` is used to implement round-robin attempt at connecting to given nameservers. Basically if an attempt to connect to a namesever fails then it is pushed to the end of the queue and we try the next one in the queue. Ordered as IPv6 ascending. https://github.com/mirage/ocaml-dns/pull/312/files#diff-f1c4d1e041cdee765597667b7e7a51894b20b99d1f5deb97adf94925f82d9c1aR152